scripts/android/register-app-version.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# This script will register a specific Android application version with Entur's
+# MobileApplicationRegistryService API.
+# The SHA256 hash of the signing key for the APK must be provided as an environment variable,
+# in addition to client ID and client secret for Entur ABT OAuth login.
+
+# Sanity check for number of parameters
+if [[ $# -ne 3 ]]; then
+    echo "Usage: <environment> <apk file> <version>"
+    exit 1
+fi
+
+app_file=$2
+app_version=$3
+
+# Check for secrets from env vars
+if [[
+    -z "${SIGNING_CERTIFICATE_HASH}"
+    || -z "${ENTUR_CLIENT_ID}"
+    || -z "${ENTUR_CLIENT_SECRET}"
+   ]]; then
+    echo "Argument error!"
+    echo "Expected three env variables:
+  - SIGNING_CERTIFICATE_HASH
+  - ENTUR_CLIENT_ID
+  - ENTUR_CLIENT_SECRET"
+    exit 2
+fi
+
+# Get values based on environment
+case $1 in
+  atb-staging)
+    authority="ATB:Authority:1"
+    app_id="app-staging"
+    token_url="https://partner-abt.staging.entur.org/oauth/token"
+    abt_url="https://core-abt-abt.staging.entur.io"
+    ;;
+  atb-prod)
+    authority="ATB:Authority:1"
+    app_id="app-store"
+    token_url="https://partner-abt.entur.org/oauth/token"
+    ;;
+
+  *)
+    echo "Unrecognized environment '$1'"
+    exit 3
+    ;;
+esac
+
+
+# Get hash for file
+if ! [[ -f $app_file ]]; then
+    echo "File '$app_file' does not exist."
+    exit 4
+fi
+
+app_hash=$(sha256sum "$app_file" | cut -d ' ' -f 1)
+
+# App login for register call
+login=$(curl --silent \
+  --request POST \
+  --url "$token_url" \
+  --header 'content-type: application/x-www-form-urlencoded' \
+  --fail-with-body \
+  --data grant_type="client_credentials" \
+  --data client_id="$ENTUR_CLIENT_ID" \
+  --data client_secret="$ENTUR_CLIENT_SECRET" \
+  --data audience="https://v2.api.entur.no")
+
+login_status=$?
+if [ $login_status -ne 0 ]; then
+    echo "Login failed: $login"
```

Comment:
Samme her

Reply:
Her har login returnert en annen status enn HTTP 200, så det burde _virkelig_ ikke være noe token der. Respons vil typisk være en melding om at brukernavn / passord ikke er gyldig. Jeg tenker at vi kan la denne stå, men kan være med på å ta den bort hvis du vil være ekstra super sikker.

Reply:
Gjerne ta bort for å være helt sikker 👍🏻

Reply:
OK, tar den bort, men du kommer til å angre deg når det blir behov for feilsøking…